parser/src/cfg/mod.rs

```diff
@@ -1076,6 +1090,24 @@ fn parse_layer_indexes(exprs: &[SpannedLayerExprs], expected_len: usize) -> Resu
         if layer_indexes.get(&layer_name).is_some() {
             bail_expr!(layer_expr, "duplicate layer name: {}", layer_name);
         }
+        let mut cfg_third = 0;
+        if let Some(third) = subexprs.peek() {
```

Comment:
You can rust-ify this to reduce if nesting with [and_then](https://doc.rust-lang.org/std/option/enum.Option.html#method.and_then).

Reply:
tried, but then the inner `subexprs.next();` stopped working due to multiborrows